File: Beecrowd/1045 Tipos de Triângulos/Main.java
import java.io.IOException;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);

        double a, b, c, temp;

        a = sc.nextDouble();
        b = sc.nextDouble();
        c = sc.nextDouble();

        if (a < b) {
            temp = a;
            a = b;
            b = temp;
        }

        if (b < c) {
            temp = b;
            b = c;
            c = temp;
        }

        if (a < b) {
            temp = a;
            a = b;
            b = temp;
        }

        if (a >= b + c)
            System.out.println("NAO FORMA TRIANGULO");
        else if (a * a == b * b + c * c)
            System.out.println("TRIANGULO RETANGULO");
        else if (a * a > b * b + c * c)
            System.out.println("TRIANGULO OBTUSANGULO");
        else if (a * a < b * b + c * c)
            System.out.println("TRIANGULO ACUTANGULO");

        if (a == b && b == c)
            System.out.println("TRIANGULO EQUILATERO");
        else if (a == b || b == c)
            System.out.println("TRIANGULO ISOSCELES");

        sc.close();
    }

}

File: Beecrowd/1045 Tipos de Triângulos/main.cpp
#include <iostream>

using namespace std;

int main() {

    double A, B, C, temp;

    cin >> A;
    cin >> B;
    cin >> C;

    if (A < B) {
        temp = A;
        A = B;
        B = temp;
    }

    if (B < C) {
        temp = B;
        B = C;
        C = temp;
    }

    if (A < B) {
        temp = A;
        A = B;
        B = temp;
    }

    if (A >= B + C)
        cout << "NAO FORMA TRIANGULO" << endl;

    else if (A * A == B * B + C * C)
        cout << "TRIANGULO RETANGULO" << endl;

    else if (A * A > B * B + C * C)
        cout << "TRIANGULO OBTUSANGULO" << endl;

    else if (A * A < B * B + C * C)
        cout << "TRIANGULO ACUTANGULO" << endl;

    if (A == B && B == C)
        cout << "TRIANGULO EQUILATERO" << endl;

    else if (A == B || B == C)
        cout << "TRIANGULO ISOSCELES" << endl;

    return 0;
}